### PR TITLE
Switch to new crossplane build module and incorporate latest uptest setup

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

Switch to new crossplane build module and incorporate latest uptest setup

I have:

- [ ] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,11 @@ XPKG_REG_ORGS_NO_PROMOTE ?= xpkg.upbound.io/upbound
 XPKGS = $(PROJECT_NAME)
 -include build/makelib/xpkg.mk
 
+CROSSPLANE_VERSION = 1.16.0-up.1
```
Comment (haarchri) [inline Makefile:30]: why we pin the version ?

Comment (ytsarev) [reply Makefile:30]: new module perk

Comment (haarchri) [reply Makefile:30]: ;( can we then update the renovate job to get new versions automatically ?

Comment (ytsarev) [reply Makefile:30]: yes, totally, let me try to setup renovate

Comment (ytsarev) [reply Makefile:30]: @haarchri please check https://github.com/upbound/configuration-azure-network/pull/44/commits/8b256f020e683cb49901490c986d926edd33e569 . Any idea how to reliably test it upfront?

Comment (haarchri) [reply Makefile:30]: You can run renovate locally against the repo folder

Comment (ytsarev) [reply Makefile:30]: @haarchri thanks for the hint added `versioningTemplate` at https://github.com/upbound/configuration-azure-network/pull/44/commits/3b37a322d3ae7d43ca198d057c0eb5dd4acb4ba8 after the local experiments

Simulated update looks working as expected:
```
LOG_LEVEL=debug GITHUB_COM_TOKEN=<token> renovate --platform=local --repository-cache=reset --dry-run=lookup
...
"depName": "upbound/universal-crossplane",
                 "currentValue": "1.15.0-up.1",
                 "datasource": "github-releases",
                 "versioning": "regex:^v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)-up.1$",
                 "extractVersion": "^v(?<version>.*)$",
                 "replaceString": "CROSSPLANE_VERSION = 1.15.0-up.1\n",
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "1.16.0-up.1",
                     "newValue": "1.16.0-up.1",
                     "releaseTimestamp": "2024-05-29T14:18:42.000Z",
                     "newMajor": 1,
                     "newMinor": 16,
                     "newPatch": 0,
                     "updateType": "minor",
                     "branchName": "renovate/upbound-universal-crossplane-1.x"
                   }
                 ],
                 "packageName": "upbound/universal-crossplane",
                 "warnings": [],
                 "sourceUrl": "https://github.com/upbound/universal-crossplane",
                 "registryUrl": "https://github.com/",
                 "currentVersion": "1.15.0-up.1",
                 "currentVersionTimestamp": "2024-02-29T15:03:05.000Z",
                 "isSingleVersion": true,
                 "fixedVersion": "1.15.0-up.1"
...
```

**Review by haarchri — APPROVED** [review]

**Pull request merged**